gfx/drivers_context/mali_fbdev_ctx.c

```diff
@@ -93,7 +93,26 @@ static int gfx_ctx_mali_fbdev_get_vinfo(void *data)
            (vinfo.yres + vinfo.upper_margin + vinfo.lower_margin + vinfo.vsync_len) /
            (vinfo.xres + vinfo.left_margin  + vinfo.right_margin + vinfo.hsync_len);
    }else{
-      mali->refresh_rate = 60;
+      //Workaround to retrieve current refresh rate if no info is available from IOCTL. If this fails as well, 60Hz is assumed...
+      FILE *fr;
```

Comment:
Can you make this use filestream instead? Can you also use a C comment here instead of a C++ comment? Thanks.

Reply:
comment fixed.

> Can you make this use filestream instead?

Do you mean looping directly with `fgetc` instead of getting the entire thing with `fgets`?
In this specific case, purpose is just parsing a small string of type `1080p60hz`, read from `/sys/class/display/mode`, to get the `60`. So it looked simpler to go with `fgets`.

Reply:
He means you should use libretro-common's filestream library.

```c
RFILE *fr = filestream_open("/sys/class/display/mode", RETRO_VFS_FILE_ACCESS_READ, RETRO_VFS_FILE_ACCESS_HINT_NONE);

if (fr)
{
   if (filestream_gets(fr, temp, sizeof(temp)))
   {
      ...
   }

   filestream_close(fr);
}
```

Alternatively, you can use the universal stream interface:
```c
intfstream_t *stream = intfstream_open_file("/sys/class/display/mode", RETRO_VFS_FILE_ACCESS_READ, RETRO_VFS_FILE_ACCESS_HINT_NONE);

if (stream)
{
   if (intfstream_gets(stream, temp, sizeof(temp)))
   {
      ...
   }

   intfstream_close(stream);
   free(stream);
}
```

Reply:
fixed, thanks @Cthulhu-throwaway.